source/common/ssl/context_impl.cc

```diff
@@ -259,6 +267,12 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const ContextConfig& config, TimeS
                                          "ECDSA certificates are supported",
                                          ctx.cert_chain_file_path_));
       }
+    } else if (have_rsa_cert) {
```

Comment:
There are other certificate types (though, they are not very popular), but this code path assumes that !ECDSA == RSA, so it would potentially reject those.